base/sort.jl

```diff
@@ -172,18 +172,20 @@ partialsort(v::AbstractVector, k::Union{Integer,OrdinalRange}; kws...) =
 # index of the first value of vector a that is greater than or equal to x;
 # returns lastindex(v)+1 if x is greater than all values in v.
 function searchsortedfirst(v::AbstractVector, x, lo::T, hi::T, o::Ordering)::keytype(v) where T<:Integer
-    u = T(1)
-    lo = lo - u
-    hi = hi + u
-    @inbounds while lo < hi - u
-        m = midpoint(lo, hi)
+    hi = hi + T(1)
+	len = hi - lo
+    @inbounds while len != 0
+		half_len = len >> 1
```

Comment:
`midpoint` already does that without accidental promotion to `Int`:

```
julia> @edit Base.midpoint(1,2)
[...]
# This implementation of `midpoint` is performance-optimized but safe
# only if `lo <= hi`.
midpoint(lo::T, hi::T) where T<:Integer = lo + ((hi - lo) >>> 0x01)
midpoint(lo::Integer, hi::Integer) = midpoint(promote(lo, hi)...)
[...]
```

What's the added benefit of using `>>` compared to `>>>` here?

Reply:
no benefit, I'm a Julia noob and didn't know it exists :)